src/summon/browzine-summon-adapter.js

```diff
@@ -205,20 +233,108 @@ browzine.summon = (function() {
   function getArticleLinkUrl(scope, data) {
     var articleLinkUrl = null;
 
-    if(isArticle(scope)) {
-      if(data && data.contentLocation) {
+    if (isArticle(scope)) {
+      if (data && data.contentLocation) {
         articleLinkUrl = data.contentLocation;
       }
     }
 
     return articleLinkUrl;
   };
 
+  function isTrustedRepository(response) {
+    var validation = false;
+
+    if (response.best_oa_location && response.best_oa_location.url_for_pdf) {
+      if (response.best_oa_location.url_for_pdf.indexOf('nih.gov') > -1 || response.best_oa_location.url_for_pdf.indexOf('europepmc.org') > -1) {
+        validation = true;
+      }
+    }
+
+    return validation;
+  };
+
+  function isUnknownVersion(response) {
+    var validation = false;
+
+    if (response.best_oa_location) {
+      if (!response.best_oa_location.version || response.best_oa_location.version === '') {
+        validation = true;
+      }
+    }
+
+    return validation;
+  };
+
+  function getUnpaywallArticlePDFUrl(response) {
+    var url;
+
+    if (response.best_oa_location) {
+      if (response.best_oa_location.host_type === "publisher" || response.best_oa_location.host_type === "repository") {
```

Comment:
> If that's the case, I wonder if we could just take this logic and bring it over to Nomad more or less copy/pasted, so that in the future when we have Unpaywall logic changes, we can put the files side-by-side and do a line-for-line modification. What do you think @tiandavis ?

That would be so GREAT Karl! It was so hard distilling everything initially. Would be good to sync them back up now that the dust has settled a bit and we have more clarity on the Unpaywall use. Will get a PR going for Nomad as part of this effort! 🙏

> Is part of the reason to not use a shared file between browzine-summon-adapter and browzine-primo-adapter here because it's difficult to do requires in these files? I see we don't have requires in here currently.

We just need a build system going, some combination of webpack and babel would be great. We wouldn't necessarily even need to minify the codebase, so we may not need business signoff...Actually, we may need business signoff because one of the initial goals was for integrators to be able to copy and paste the code. But...with our automated deploy and update system, the benefit of integrators getting live updates far far far outweighs the copy-and-paste benefit. So maybe we just need buy in and priority, then we can probably get a build system going which would allow for more code reuse, etc.

Reply:
> Will get a PR going for Nomad as part of this effort! 🙏

Excellent! I think it's worth it - and if you make that PR, then we can double-check the diff on that repo to confirm all the logic matches up.

> one of the initial goals was for integrators to be able to copy and paste the code.

Ooh, good point, I didn't know if that was the case. But yeah, I think copy-and-paste probably isn't used much. Maybe check with John or Sarah in Slack?

Reply:
https://thirdiron.slack.com/archives/C6ND59F7C/p1582927984006600

Reply:
Nomad PR:
https://github.com/thirdiron/nomad/pull/110